2.arrayOccurenceFilter.js

```diff
@@ -11,7 +11,24 @@
 // occurrenceFilter([1, 3, 4, 1, 9, 1, 3, 4, 3, 1, 2], 2) returns [1, 3, 4] 
 
 const occurrenceFilter = (arr, n) => {
+    let result = [];
+    let seen = {};
 
+    while(arr.length) {
+        let currentNum = arr[0];
+        if(seen[currentNum]) {
+            seen[currentNum] += 1
+            if(seen[currentNum] === n) {
+                result.push(currentNum)
+            }
+        } else {
+            seen[currentNum] = 1;
+        }
+
+        arr.shift();
```

Comment:
Generally avoid mutating the input unless needed. A simple forEach loop would suffice.